src/SQLStore/PropertyTableInfoFetcher.php

```diff
@@ -0,0 +1,219 @@
+	 *
+	 * @var TableDefinition[]|null
+	 */
+	private $propertyTableDefinitions = null;
+
+	/**
+	 * Array to cache "propkey => table id" associations for fixed property
+	 * tables. Initialized by getPropertyTables(), which must be called
+	 * before accessing this.
+	 *
+	 * @var array|null
+	 */
+	private $fixedPropertyTableIds = null;
+
+	/**
+	 * Keys of special properties that should have their own
+	 * fixed property table.
+	 *
+	 * @var array
+	 */
+	private $customizableSpecialProperties = array(
+		'_MDAT', '_CDAT', '_NEWP', '_LEDT', '_MIME', '_MEDIA',
+	);
+
+	/**
+	 * @var array
+	 */
+	private $customSpecialPropertyList = array();
+
+	/**
+	 * @var array
+	 */
+	private $fixedSpecialProperties = array(
+		// property declarations
+		'_TYPE', '_UNIT', '_CONV', '_PVAL', '_LIST', '_SERV',
+		// query statistics (very frequently used)
+		'_ASK', '_ASKDE', '_ASKSI', '_ASKFO', '_ASKST', '_ASKDU',
+		// subproperties, classes, and instances
+		'_SUBP', '_SUBC', '_INST',
+		// redirects
+		'_REDI',
+		// has sub object
+		'_SOBJ',
+		// vocabulary import and URI assignments
+		'_IMPO', '_URI',
+		// Concepts
+		'_CONC'
+	);
+
+	/**
+	 * @var array
+	 */
+	private $customFixedPropertyList = array();
+
+	/**
+	 * Default tables to use for storing data of certain types.
+	 *
+	 * @var array
+	 */
+	private $defaultDiTypeTableIdMap = array(
+		DataItem::TYPE_NUMBER     => 'smw_di_number',
+		DataItem::TYPE_BLOB       => 'smw_di_blob',
+		DataItem::TYPE_BOOLEAN    => 'smw_di_bool',
+		DataItem::TYPE_URI        => 'smw_di_uri',
+		DataItem::TYPE_TIME       => 'smw_di_time',
+		DataItem::TYPE_GEO        => 'smw_di_coords', // currently created only if Semantic Maps are installed
+		DataItem::TYPE_WIKIPAGE   => 'smw_di_wikipage',
+		//DataItem::TYPE_CONCEPT    => '', // _CONC is the only property of this type
+	);
+
+	/**
+	 * @since 2.2
+	 *
+	 * @param array $customFixedProperties
+	 */
+	public function setCustomFixedPropertyList( array $customFixedProperties ) {
+		$this->customFixedPropertyList = $customFixedProperties;
+	}
+
+	/**
+	 * @since 2.2
+	 *
+	 * @param array $customSpecialProperties
+	 */
+	public function setCustomSpecialPropertyList( array $customSpecialProperties ) {
+		$this->customSpecialPropertyList = $customSpecialProperties;
+	}
+
+	/**
+	 * Find the id of a property table that is suitable for storing values of
+	 * the given type. The type is specified by an SMW type id such as '_wpg'.
+	 * An empty string is returned if no matching table could be found.
+	 *
+	 * @since 1.8
+	 *
+	 * @param string $dataTypeTypeId
+	 *
+	 * @return string
+	 */
+	public function findTableIdForDataTypeTypeId( $dataTypeTypeId ) {
+		return $this->findTableIdForDataItemTypeId(
+			DataTypeRegistry::getInstance()->getDataItemId( $dataTypeTypeId )
+		);
+	}
+
+	/**
+	 * Find the id of a property table that is normally used to store
+	 * data items of the given type. The empty string is returned if
+	 * no such table exists.
+	 *
+	 * @since 1.8
```

Comment:
This method, `PropertyTableInfoFetcher::findTableIdForDataItemTypeId`, only exists since this commit